### DISCUSSION TITLE
My backup is encrypted, is this supported?

### DISCUSSION THREAD

**Discussion by badgersmans** [ReagentX/imessage-exporter#320] [Q&A] [answered]
As the title says; I have the backup password so decrypting it is no problem, but i'm not sure how to decrypt it and make it work with imessage-exporter. Any help would be greatly appreciated, thank you all!

**Comment by ReagentX** [comment]
~~No, encrypted backups are not supported. You would need to make an unencrypted backup.~~

~~There do appear to be several open source projects that can decrypt iOS backups, but I have not worked with any of them.~~

**Reply by ReagentX** [reply, marked as answer]
Support for encrypted devices was added in the [Canyon Sunflower](https://github.com/ReagentX/imessage-exporter/releases/tag/2.7.1) release.

**Comment by badgersmans** [comment]
I have used https://github.com/PeterUpfold/dump-iphone-backup?tab=readme-ov-file to decrypt the backup and got a bunch of folders with names like AppDomain-com.apple.PASViewService..i wondering if it's possible to somehow make use of all these folders? 

And I can't make another unencrypted backup because i have since factory reset my iphone..

**Reply by ReagentX** [reply]
Is there anything named `3d0d7e5fb2ce288813306e4d4636395e047a3d28` inside of the results? If so, that is the messages database file.

**Comment by badgersmans** [comment]
well searching for that using my mac's finder didn't seem to bring up anything..however i know my messages has lots of media, and so sorting by folder sizes i see this being the biggest folder, it has the following structure:

<img width="295" alt="image" src="https://github.com/user-attachments/assets/1bfeb583-3230-4714-9e83-8b74f637e590">

**Reply by badgersmans** [reply]
I mean lots of folders, something like this:

<img width="1453" alt="image" src="https://github.com/user-attachments/assets/7676583b-0b4b-4829-a05e-a1875a48834e">

**Reply by ReagentX** [reply]
The database should be located in `Library/SMS/sms.db`, and you may be able to point `imessage-exporter` at it using the `-p` flag.

**Reply by badgersmans** [reply]
oh no 😢  it's hopeless, i don't think it is anywhere. i tried pointing to the sms.db located at /HomeDomain/Library/SMS/sms.db (the only one found by finder).

And the 3d0d7e5fb2ce288813306e4d4636395e047a3d28 is not found anywhere by finder. Guess if I really want to, i could wipe my phone and restore from this backup and then from here create an unencrypted backup! Emm..yea i guess this can be resolved. Thank you so much kind Sir!

**Reply by ReagentX** [reply]
Your encrypted backup does have the sms database in it; the software you used probably didn't extract it properly or put it in a different place. I would try a different project and see if you get a different result.
